File: gamerank-backend/src/main/java/com/BrunoDev/gamerank_backend/model/Game.java
package com.BrunoDev.gamerank_backend.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "games")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Game {
    @Id
    private String id;
    private String title;
    private String genre;
    private String developer;
    private List<Review> reviews;
}

File: gamerank-frontend/src/app/models/game.ts
export interface Review {
  id: string;
  gameId: string;
  author: string;
  content: string;
  rating: number;
  date: string | number[];
}

export interface Game {
  id: string;
  title: string;
  genre: string;
  developer: string;
  reviews: Review[];
}
